version-parser/version-parser.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/python
+
+################################################################################
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+import sys
+import commands
+import os
+
+major = None
+minor = None
+security = None
+build = None
+opt = None
+semver = None
+version = None
+
+if "TEST" in os.environ:
+    output = os.environ['TEST']
+    build_num = sys.argv[2]
+else:
+    java_cmd = sys.argv[1]
+    build_num = sys.argv[2]
+    status, output = commands.getstatusoutput(java_cmd + ' -version')
+
+# version_string should look like OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.2+7)
+version_string = output.split('\n', 1)[1]
+
+# returns a string like 1.8.0_202-b08 or 11.0.2+7
+version = version_string.split('build ')[1].split(')')[0]
+
+split_version = version.split('.')
+
+try:
+    isinstance(int(split_version[0]), (int, long))
+except ValueError:
+    split_version = version.split('+')
+
+if int(split_version[0]) > 1:
+    # detected openjdk 9 or above
+    major = int(split_version[0]) # 11
+    try:
+        minor = int(split_version[1]) # 0
```

Comment:
does this work for `11+28`?

Reply:
tested here: https://github.com/AdoptOpenJDK/openjdk-build/blob/21cd7cf4162f444fa9ceb95c87f3f71870dd8bca/version-parser/test-version-parser.sh#L68-L70